src/main/java/frc/robot/subsystems/arm/ArmExtSubsystem.java

```diff
@@ -0,0 +1,224 @@
+package frc.robot.subsystems.arm;
+
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.SparkMaxPIDController;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import lib.factories.SparkMaxFactory;
+import lib.utils.drivers.RevUtil;
+import org.littletonrobotics.junction.AutoLog;
+import org.littletonrobotics.junction.Logger;
+
+public class ArmExtSubsystem extends SubsystemBase {
+    private final CANSparkMax m_armExt;
+    private final RelativeEncoder m_relativeEncoderArmEx;
+    private final SparkMaxPIDController m_extPID;
+//    private final PIDController m_extPID;
+    private final DigitalInput m_armLimitSwitch;
+
+    private final ArmExtIOInputsAutoLogged m_inputs = new ArmExtIOInputsAutoLogged();
+    // Logging variables
+    private double prevSetpointRaw;
+    private double prevSetpointClamped;
+    public static boolean m_hasArmHomed = false;
+    //Shuffleboard data
+    private final ShuffleboardTab armExtTab;
+    private GenericEntry armExtAtSetpointEntry;
+    private GenericEntry armExtAtUpperLimitEntry;
+    private GenericEntry armExtAtLowerLimitEntry;
+    private GenericEntry armExtConvertedEntry;
+    private GenericEntry armExtTargetEntry;
+    private GenericEntry armExtSetpointClampedEntry;
+    private GenericEntry armExtMotorOutputEntry;
+
+
+
+
+
+    @AutoLog
+    public static class ArmExtIOInputs {
+        public double armExtension = 0.0;
+    }
+
+    public ArmExtSubsystem() {
+        SparkMaxFactory.SparkMaxConfig config = new SparkMaxFactory.SparkMaxConfig();
+        config.setCurrentLimit(50);
+        config.setInverted(Constants.CURRENT_MODE == Constants.Mode.HELIOS_V1);
+
+        m_armExt = SparkMaxFactory.Companion.createSparkMax(Constants.ArmConstants.ARM_EXTENSION_ID, config);
+        m_armExt.setClosedLoopRampRate(0.1);
+
+        m_relativeEncoderArmEx = m_armExt.getEncoder();
+        RevUtil.autoRetry(() -> m_relativeEncoderArmEx.setPositionConversionFactor(Constants.ArmConstants.EXTENSION_RATIO));
+
+        m_extPID = m_armExt.getPIDController();
+        m_extPID.setP(Constants.ArmConstants.ARM_EXT_KP.getValue());
+        m_extPID.setI(Constants.ArmConstants.ARM_EXT_KI.getValue());
+        m_extPID.setD(Constants.ArmConstants.ARM_EXT_KD.getValue());
```

Comment:
Not utilizing "tunability". See the [rev wrapper ](https://github.com/GirlsOfSteelRobotics/GirlsOfSteelFRC/blob/main/libraries/GirlsOfSteelLibRev/src/main/java/com/gos/lib/rev/RevPidPropertyBuilder.java)